### ISSUE TITLE
The view 'Index' was not found

### ISSUE THREAD

**Issue by Genbox** [ExtCore/ExtCore#22]
I noticed that when I set the output path of an assembly, the views no longer load.

Steps to reproduce:
1. Download the ready to use sample at http://extcore.net/files/ExtCore-Sample-1.0.0-alpha16.zip
2. Change line 3 in ExtCore-Sample\src\WebApplication.ExtensionA\project.json from
`"buildOptions": { "embed": [ "Styles/**", "Views/**" ],`
To:
`"buildOptions": { "embed": [ "Styles/**", "Views/**" ], "outputName": "Something.ExtensionA"},`
3. Remove WebApplication.ExtensionA.dll from the Extensions folder of WebApplication
4. Move Something.ExtensionA.dll from the output folder of the WebApplication.ExtensionA project into the Extensions folder in WebApplication

You should receive an error like this:
InvalidOperationException: The view 'Index' was not found. The following locations were searched:
/Views/ExtensionA/Index.cshtml
/Views/Shared/Index.cshtml

**Comment by DmitrySikorsky** [comment]
Hm, I think it might be related to something like https://github.com/ExtCore/ExtCore/blob/master/src/ExtCore.Mvc/MvcExtension.cs#L122

Maybe a.GetName().Name returns 'WebApplication.ExtensionA' so resources will have default namespace 'WebApplication.ExtensionA', not 'Something.ExtensionA', but I'm not sure so far.

**Comment by DmitrySikorsky** [comment]
I have explored this issue and found that when you set outputName to any value your embedded resources still have original names (prefix = project name). So in this case line https://github.com/ExtCore/ExtCore/blob/master/src/ExtCore.Mvc/MvcExtension.cs#L122 will set wrong base namespace so resources will not be discovered. To fix that we need to be able to get original namespace from the assembly. I don't know how to do that so far.

**Comment by mwgriffiths88** [comment]
Ideally grabbing it from the Assembly would be ideal. However, I can't find anything that would work.

It might be possible to load the correct namespace by getting access to the extension's `Startup` class and calling `this.GetType().GetTypeInfo()`. It has properties for `Assembly` and `Namespace` which would work for this scenario - I'll investigate a little later on.

**Comment by DmitrySikorsky** [comment]
Problem is that assembly may have only resources, without any classes. So still don't know how to fix. Maybe we should ask on asp.net core project.

**Comment by mwgriffiths88** [comment]
As far as I can tell the [EmbeddedFileProvider](https://github.com/aspnet/FileSystem/blob/dev/src/Microsoft.Extensions.FileProviders.Embedded/EmbeddedFileProvider.cs) works by taking a path like '/Views/Home/Index.cshtml' and transforming it to 'Views.Home.Index.cshtml', and then appending the `baseNamespace`, like 'WebApplication.ExtensionA'.

It's possible that we could our own provider that simply queries the `GetManifestResourceNames()` inside the assembly for the correct resource. The only problem is that there would be a conflict between extensions that contain the same view in the same location - but this problem must already exist with the current setup too?

**Comment by DmitrySikorsky** [comment]
Yes, there will be a conflict if we have the same view with the same location in the different extensions. And I think it is normal, because the point is that you can specify the views for the extensions inside your main web application of inside the different extensions... It is the way it should work as I think. So you solution may work.

**Comment by bdparrish** [comment]
Should this issue be closed?

**Comment by mwgriffiths88** [comment]
The issue still exists, the submitted solution was a workaround and nothing more. I'm happy for the issue to be closed.

**Comment by DmitrySikorsky** [comment]
Agree with @mwgriffiths88. I will be happy to know how to fix the issue (not workaround).
